icu4c/source/common/dictbe.cpp

```diff
@@ -1341,6 +1353,20 @@ CjkBreakEngine::divideUpDictionaryRange( UText *inText,
     if ((uint32_t)bestSnlp.elementAti(numCodePts) == kuint32max) {
         t_boundary.addElement(numCodePts, status);
         numBreaks++;
+    } else if (isPhraseBreaking) {
```

Comment:
FYI
1. The logic here looks plausible, and makes sense given the implementation doc, but I am not familiar with the BreakIterator implementation, and I trust Frank and Andy for having helped get this right. Praise to Frank & Andy!
2. Praise to the implementers: I am very impressed by how little code is necessary for this feature!